app/controllers/application_controller.rb

```diff
@@ -77,15 +77,34 @@ def sign_in(login_userinfo)
 
     @current_user = user
     renew_session
+
+    SecurityLog.log_event(
+      action: "accessed_site",
+      originator: @current_user,
+      remote_ip: request.remote_ip
+    )
+
     session[:userinfo] = login_userinfo
   end
 
   def sign_out
+    last_login = SecurityLog.where(originator: @current_user, action: "accessed_site").order(logged_at: :desc).first
+
+    duration = if last_login
+                 Time.current - last_login.logged_at
+               end
```

Comment:
it looks like the Phoenix implementation stores the duration as an integer but this stores a float. I think we should convert to to be consistent.